assets/js/modules/analytics/components/common/SetupUseSnippetSwitch.js

```diff
@@ -34,10 +34,37 @@ export default function SetupUseSnippetSwitch() {
 	const existingTag = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS ).getExistingTag()
 	);
+
+	const canUseSnippet = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getCanUseSnippet()
+	);
+
 	const propertyID = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS ).getPropertyID()
 	);
 
+	if (
+		! existingTag &&
+		( canUseSnippet === undefined || canUseSnippet === true )
+	) {
+		return null;
+	}
```

Comment:
I think I wrote this point in IB incorrectly. The Component should render if either of the following conditions are correct:

* There's an existing tag. 
* There's no existing tag, the canUseSnippet is false. 

Based on the above conditions, the logic I wrote makes sense, however, is it too implicit and we should clarify it with nested if/else? 

Cheers.

Reply:
Looks good to me, however we want to render the component if ``canUseSnippet` is `false`. Will the following work? 

```js
if ( ! ( Boolean( existingTag ) || canUseSnippet === true ) ) {
	return null;
}
```

I left the `undefined` check here as it was there before the ETS as well. Can get rid of it if that will cause no issue.

Reply:
> Looks good to me, however we want to render the component if `canUseSnippet` is `false`

That is what I had above – `if ! conditions_that_should_render then return null` with conditions that should render being: any existing tag, or canUseSnippet set to `false`, so changing that to `true` would cause it to return `null` due to the wrapping negation

Reply:
Ahhh. I didn't realize the negation was wrapping the entire thing.